skyportal/handlers/api/analysis.py

```diff
@@ -0,0 +1,423 @@
+                        should contain the key `authentication_type`. Values of this key will be used
+                        to POST into the remote analysis service.
+                  enabled:
+                    type: boolean
+                    description: Whether the service is enabled or not.
+                  type:
+                    type: string
+                    description: Type of analysis. One of: {', '.join(f"'{t}'" for t in ANALYSIS_TYPES)}
+                  input_data_types:
+                    type: array
+                    items:
+                        type: string
+                    description: |
+                        List of input data types that the service requires. Zero to many of:
+                        {', '.join(f"'{t}'" for t in ANALYSIS_INPUT_TYPES)}
+                  timeout:
+                    type: float
+                    description: Max time in seconds to wait for the analysis service to complete. Defult is 3600.0.
+                    default: 3600.0
+                  group_ids:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups should be
+                      able to use the Analysis Service. Defaults to all of requesting
+                      user's groups.
+                required:
+                  - name
+                  - url
+                  - authentication_type
+                  - type
+                  - input_data_types
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New AnalysisService ID
+        """
+        data = self.get_json()
+
+        if not data.get('url', None):
+            return self.error('`url` is required to add an Analysis Service.')
+        else:
+            if not valid_url(data.get('url', None)):
+                return self.error(
+                    'a valid `url` is required to add an Analysis Service.'
+                )
+
+        authentication_type = data.get('authentication_type', None)
+        if not authentication_type:
+            return self.error(
+                '`authentication_type` is required to add an Analysis Service.'
+            )
+
+        if authentication_type not in AUTHENTICATION_TYPES:
+            return self.error(
+                f'`authentication_type` must be one of: {", ".join([t for t in AUTHENTICATION_TYPES])}.'
+            )
+        else:
+            if authentication_type != 'none':
```

Comment:
It might be nice to summarize in a doc/ how this is expected to work.

Reply:
There will be more complete description of this in the next PR (it's where we actually implement the log in).